### DISCUSSION TITLE
Nextcloud 4.9.0 - SSL issue (Collabora and Talk)

### DISCUSSION THREAD

**Discussion by elephantastyczny** [nextcloud/all-in-one#2425] [Questions] [answered]
Good morning everyone.
I run into issue and I cannot deal with it by myself, hence cry for help.

I had flawlessly running Nextcloud AIO and decided to add Reverse Proxy to the setup.
So I installed NPM (used setup from https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md)  forwarded ports on Mikrotik router, but I couldn't get it working at all, so after that I reverted configuration on router back to previous one and since then I had issues to display nextcloud web page (https:).
AIO interface was always full green.

![image](https://user-images.githubusercontent.com/120596179/234825118-83f30e4c-ea14-46c0-9cbd-376142961e17.png)

Of course I tried to restore data from backup but it didn't help.

So what I've done next was to add Caddy to docker compose.

`
version: "3.8"

volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer # This line is not allowed to be changed

services:
  nextcloud:
    image: nextcloud/all-in-one:latest
    restart: always
    container_name: nextcloud-aio-mastercontainer # This line is not allowed to be changed
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config # This line is not allowed to be changed
      - /var/run/docker.sock:/var/run/docker.sock:ro # May be changed on macOS, Windows or docker rootless. See the applicable documentation. If adjusting, don't forget to also set 'DOCKER_SOCKET_PATH'!
    ports:
    #   - 80:80 # Can be removed when running behind a web server or reverse proxy (like Apache, Nginx and else). See https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md
      - 8080:8080
    #   - 8443:8443 # Can be removed when running behind a web server or reverse proxy (like Apache, Nginx and else). See https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md
    environment: # Is needed when using any of the options below
      - APACHE_PORT=11000 # Is needed when running behind a web server or reverse proxy (like Apache, Nginx and else). See https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md
      - APACHE_IP_BINDING=0.0.0.0 # Should be set when running behind a web server or reverse proxy (like Apache, Nginx and else) that is running on the same host. See https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md
      # - COLLABORA_SECCOMP_DISABLED=false # Setting this to true allows to disable Collabora's Seccomp feature. See https://github.com/nextcloud/all-in-one#how-to-disable-collaboras-seccomp-feature
      # - DOCKER_SOCKET_PATH=/var/run/docker.sock # Needs to be specified if the docker socket on the host is not located in the default '/var/run/docker.sock'. Otherwise mastercontainer updates will fail. For macos it needs to be '/var/run/docker.sock'
      # - DISABLE_BACKUP_SECTION=false # Setting this to true allows to hide the backup section in the AIO interface.
      # - NEXTCLOUD_DATADIR=/mnt/ncdata # Allows to set the host directory for Nextcloud's datadir. See https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir
      # - NEXTCLOUD_MOUNT=/mnt/ # Allows the Nextcloud container to access the chosen directory on the host. See https://github.com/nextcloud/all-in-one#how-to-allow-the-nextcloud-container-to-access-directories-on-the-host
      - NEXTCLOUD_UPLOAD_LIMIT=10G # Can be adjusted if you need more. See https://github.com/nextcloud/all-in-one#how-to-adjust-the-upload-limit-for-nextcloud
      - NEXTCLOUD_MAX_TIME=3600 # Can be adjusted if you need more. See https://github.com/nextcloud/all-in-one#how-to-adjust-the-max-execution-time-for-nextcloud
      - NEXTCLOUD_MEMORY_LIMIT=2048M # Can be adjusted if you need more. See https://github.com/nextcloud/all-in-one#how-to-adjust-the-php-memory-limit-for-nextcloud
      - NEXTCLOUD_TRUSTED_CACERTS_DIR=/opt/certs # CA certificates in this directory will be trusted by the OS of the nexcloud container (Useful e.g. for LDAPS) See See https://github.com/nextcloud/all-in-one#how-to-trust-user-defiend-certification-authorities-ca
      # - NEXTCLOUD_STARTUP_APPS=deck twofactor_totp tasks calendar contacts # Allows to modify the Nextcloud apps that are installed on starting AIO the first time. See https://github.com/nextcloud/all-in-one#how-to-change-the-nextcloud-apps-that-are-installed-on-the-first-startup
      # - NEXTCLOUD_ADDITIONAL_APKS=imagemagick # This allows to add additional packages to the Nextcloud container permanently. Default is imagemagick but can be overwritten by modifying this value. See https://github.com/nextcloud/all-in-one#how-to-add-os-packages-permanently-to-the-nextcloud-container
      # - NEXTCLOUD_ADDITIONAL_PHP_EXTENSIONS=imagick # This allows to add additional php extensions to the Nextcloud container permanently. Default is imagick but can be overwritten by modifying this value. See https://github.com/nextcloud/all-in-one#how-to-add-php-extensions-permanently-to-the-nextcloud-container
      # - NEXTCLOUD_ENABLE_DRI_DEVICE=true # This allows to enable the /dev/dri device in the Nextcloud container which is needed for hardware-transcoding. See https://github.com/nextcloud/all-in-one#how-to-enable-hardware-transcoding-for-nextcloud
      - TALK_PORT=3478 # This allows to adjust the port that the talk container is using.
      - sig-proxy=false

  caddy:
    image: caddy:alpine
    restart: always
    container_name: caddy
    volumes:
      - ./Caddyfile:/etc/caddy/Caddyfile
      - ./certs:/certs
      - ./config:/config
      - ./data:/data
      - ./sites:/srv
    ports:
     - 80:80
     - 443:443
    network_mode: "host"
`
    
Unforunately it was telling me that there is a thottling on my cert renew and I cannot get it now
    
    `ERR ts=1682582488.66889 logger=tls.obtain msg=could not get certificate from issuer identifier=**redacted** issuer=acme-v02.api.letsencrypt.org-directory error=HTTP 429 urn:ietf:params:acme:error:rateLimited - Error creating new order :: too many certificates (5) already issued for this exact set of domains in the last 168 hours: **redacted**, retry after 2023-04-28T05:33:06Z: see https://letsencrypt.org/docs/duplicate-certificate-limit/`
    
As this is a production site, I figured it out that most probably NPM also asked for certificate so I copied it to Caddyfile
`
https://**redacted**:443 {
    header Strict-Transport-Security max-age=31536000;
    reverse_proxy 192.168.0.19:11000
    tls /certs/fullchain1.pem /certs/privkey1.pem
`

This allowed me to run Nextcloud finally but I still have issues with:
Collabora
![image](https://user-images.githubusercontent.com/120596179/234827695-9b08c04f-48e3-481e-b682-19fa0c945952.png)
![image](https://user-images.githubusercontent.com/120596179/234828364-05dc92c8-5737-4360-9a9e-da448c3e0f48.png)

root@**redacted**:/data/compose/2# sudo docker exec -it nextcloud-aio-nextcloud bash
bash-5.1# curl -vvv **redacted**:443/hosting/discovery
*   Trying ip.address.redated:443...
* connect to ip.address.redated port 443 failed: Connection refused
* Failed to connect to **redacted** port 443 after 22 ms: Connection refused
* Closing connection 0
curl: (7) Failed to connect to **redacted** port 443 after 22 ms: Connection refused

Talk
![image](https://user-images.githubusercontent.com/120596179/234828037-8e641025-2952-483c-ad6a-f57f8f4643e3.png)

bash-5.1# curl -vvv https://**redacted**::443/standalone-signaling/api/v1/welcome
*   Trying 109.241.108.98:443...
* connect to 109.241.108.98 port 443 failed: Connection refused
* Failed to connect to **redacted** port 443 after 20 ms: Connection refused
* Closing connection 0
curl: (7) Failed to connect to **redacted** port 443 after 20 ms: Connection refused

![image](https://user-images.githubusercontent.com/120596179/234843925-c7c432b2-7ad1-4705-94e9-02b3655ece17.png)

I really need some assistance here guys.
Thanks!

**Comment by elephantastyczny** [comment]
Is there anything else I have to enter into caddyfile like websockets?

**Comment by elephantastyczny** [comment]
Last question before I'll stop as I'm running out of ideas.

Maybe it will be enough to repull apache, nextcloud and talk containters?

**Comment by szaimen** [comment]
> curl -vvv **redacted**:443/hosting/discovery
> 
> * Trying ip.address.redated:443...
> * connect to ip.address.redated port 443 failed: Connection refused
> * Failed to connect to **redacted** port 443 after 22 ms: Connection refused
> * Closing connection 0
>   curl: (7) Failed to connect to **redacted** port 443 after 22 ms: Connection refused

If your public ip-address is used above for the connection (and not a private one), you need to debug why the connection fails. Hint: It is caused by your network infrastucture. 

If the ip-address was a private one, see the first step in https://github.com/nextcloud/all-in-one/discussions/1358

**Comment by elephantastyczny** [comment]
Hi @szaimen thanks for reply.

Redacted address is a public one.
If those are a network issues, shouldn't I be able to run Nextcloud at all?

AIO starts without issues, Nextcloud also (I'm able to log in) Issues are with signalling backend and starting Collabora.

It is caused by your network infrastucture -> are you talking about forwarding configuration?

**Reply by szaimen** [reply]
No, sometimes the router is too cheap to support accessing the piblic domain locally. Feel free to goofle a bit around also in the threads here.

**Reply by elephantastyczny** [reply]
I'm using CCR1009-7G-1C-1S+ from Mikrotik it's more than capable to do all necessary stuff.
And trust me I'm googling as crazy ;)

**Reply by szaimen** [reply, marked as answer]
Possibly also your firewall blocks these requests...

**Reply by elephantastyczny** [reply]
You were almost 100% correct, I had to adjust my firewall configuration and setup proper NAT hairpin for VLAN on Mikrotik router.
Then restarting Carry container and voila, everything is working as it should.
